components/style/properties/gecko.mako.rs

```diff
@@ -2353,6 +2353,48 @@ fn static_assert() {
         longhands::box_shadow::computed_value::T(buf)
     }
 
+    pub fn set_clip(&mut self, v: longhands::clip::computed_value::T) {
+        use gecko_bindings::structs::NS_STYLE_CLIP_AUTO;
+        use gecko_bindings::structs::NS_STYLE_CLIP_RECT;
+        use gecko_bindings::structs::NS_STYLE_CLIP_RIGHT_AUTO;
+        use gecko_bindings::structs::NS_STYLE_CLIP_BOTTOM_AUTO;
+        use values::Either;
+
+        match v {
+            Either::First(rect) => {
+                self.gecko.mClipFlags = NS_STYLE_CLIP_RECT as u8;
+                self.gecko.mClip.x = rect.left.0;
+                self.gecko.mClip.y = rect.top.0;
+
+                if rect.bottom.is_none() {
```

Comment:
`if let Some(bottom) = self.bottom`, or `match` instead of `is_none() and then unwrap? here and below.